palace.pyx

```diff
@@ -1208,6 +1214,118 @@ cdef class SourceGroup:
         """
         self.impl.destroy()
 
+
+cdef class AuxiliaryEffectSlot:
+    """An effect processor.
+
+    It takes the output mix of zero or more sources,
+    applies DSP for the desired effect (as configured
+    by a given `Effect` object), then adds to the output mix.
+
+    This can be used as a context manager that calls `destroy`
+    upon completion of the block, even if an error occurs.
+
+    Parameters
+    ----------
+    context : Context
+        The context from which the auxiliary effect slot is to be created.
+
+    Raises
+    ------
+    RuntimeError
+        If the effect slot can't be created.
+    """
+    cdef alure.AuxiliaryEffectSlot impl
+
+    def __init__(self, context: Context) -> None:
+        self.impl = (<Context> context).impl.create_auxiliary_effect_slot()
+
+    def __enter__(self) -> AuxiliaryEffectSlot:
+        return self
+
+    def __exit__(self, exc_type: Optional[Type[BaseException]],
+                 exc_val: Optional[BaseException],
+                 exc_tb: Optional[TracebackType]) -> Optional[bool]:
+        self.destroy()
+
+    def __lt__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl < (<AuxiliaryEffectSlot> other).impl
+
+    def __le__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl <= (<AuxiliaryEffectSlot> other).impl
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl == (<AuxiliaryEffectSlot> other).impl
+
+    def __ne__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl != (<AuxiliaryEffectSlot> other).impl
+
+    def __gt__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl > (<AuxiliaryEffectSlot> other).impl
+
+    def __ge__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl >= (<AuxiliaryEffectSlot> other).impl
+
+    def __bool__(self) -> bool:
+        return <boolean> self.impl
+
+    def set_gain(self, value: float) -> None:
+        self.impl.set_gain(value)
+
+    def set_send_auto(self, value: bool) -> None:
+        self.impl.set_send_auto(value)
+
+    gain = property(fset=set_gain, doc=(
+        """Set gain for the effect slot.
+        """))
+    send_auto = property(fset=set_send_auto, doc=(
+        """If set to `True`, the reverb effect will automatically
+        apply adjustments to the source's send slot gains based
+        on the effect properties.
+
+        Has no effect when using non-reverb effects. Default is `True`.
```

Comment:
Please use double space after period.  I personally find it's easier to read fixed-width text that way, but mostly it's for overall consistency.  I certainly missed this reviewing @9a24f0's last time, so it'd be great if you can do the fix of all of it in a separate commit (I plan to squash all other ones in this PR into one).  If you plan to do mass substitution, be careful with the `e.g.` and `i.e.`.

BTW I think it'd be great if you can move the assignments down to end of class definition.  If you disagree (I have no argument to back this), feel free to file an issue discussing this.

Reply:
You mean the `TODO`s? I find none of them (including the ones made before I did anything) at the end of the class.

Reply:
I mean the property assignments, but I don't hold a very strong position on this matter.

Reply:
Well, I agree with you then.